src/main/kotlin/com/github/psxpaul/stream/OutputStreamLogger.kt

```diff
@@ -9,12 +9,16 @@ import java.io.OutputStream
 class OutputStreamLogger(private val logger: Logger) : OutputStream() {
 
     var sb = StringBuilder()
+    private var wasCR = false
 
     override fun write(b: Int) {
         val character = b.toChar()
-        if (character == '\n') {
+        if (wasCR || character == '\n') {
             logger.lifecycle(sb.toString())
```

Comment:
Would it make sense to instead remove any ending \r character here? And maybe only do that if the System.lineSeparator() is \r\n.